main.rb

```diff
@@ -0,0 +1,275 @@
+    if block_given? == false && (arr.my_all? { |x| x.nil? || x == false }) == false
+      true
+    else
+      false
+    end
+  elsif block_given? == false && !input.nil?
+    count = my_any_option(input, arr)
+    count >= 1
+  elsif block_given? == true
+    arr = to_a
+    count = 0
+    arr.my_each do |x|
+      count += 1 if yield(x) == true
+    end
+    count >= 1
+  end
+end
+
+def my_none_option(input, arr)
+  count = 0
+  arr.my_each do |x|
+    if input.is_a?(Integer)
+      count += 1 if x == input
+    elsif input.is_a?(Regexp)
+      count += 1 unless (x =~ input).nil?
+    elsif input.is_a?(Class)
+      count += 1 if x.is_a?(input)
+    end
+  end
+  count
+end
+
+def my_none?(input = nil)
+  arr = to_a
+  if block_given? == false && input.nil?
+    if block_given? == false && arr.my_all? { |x| x.nil? || x == false } == true
+      true
+    else
+      false
+    end
+  elsif block_given? == false && !input.nil?
+    count = my_none_option(input, arr)
+    count.zero?
+  elsif block_given? == true
+    arr = to_a
+    count = 0
+    arr.my_each do |x|
+      count += 1 if yield(x) == true
+    end
+    count.zero?
+  end
+end
+
+def my_count_option(input, arr)
+  count = 0
+  arr.my_each do |x|
+    if input.is_a?(Integer)
+      count += 1 if x == input
+    else
+      count
+    end
+  end
+  count
+end
+
+def my_count(input = nil)
+  if block_given? == false && input.nil?
+    arr = to_a
+    arr.length
+  elsif block_given? == false && !input.nil?
+    arr = to_a
+    count = my_count_option(input, arr)
+  elsif block_given? == true
+    count = 0
+    arr = to_a
+    arr.my_each do |x|
+      count += 1 if yield(x)
+    end
+    count
+  end
+end
+
+def my_map(input = nil)
+  return to_enum if block_given? == false && input.nil?
+
+  if block_given? && input.nil?
+    new_array = []
+    my_each do |x|
+      new_value = yield(x)
+      new_array << new_value
+    end
+  elsif block_given? == flase && !input.nil?
+    new_array = []
+    my_each do |x|
+      new_array << input[x] if input.is_a?(Proc)
+    end
+  end
+
+  new_array
+end
+
+def my_inject_option1(initial, arr)
+  if initial.is_a?(Symbol) && arr.my_all?(Integer)
+    if initial == :+
+      memo = 0
+      arr.each { |x| memo += x }
+    elsif initial == :-
+      memo = 0
+      arr.each { |x| memo -= x }
+    elsif initial == :*
+      memo = 1
+      arr.each { |x| memo *= x }
+    elsif initial == :/
+      memo = 0
+      return memo = 1 if arr.my_all?(1)
+    end
+  end
+  memo
+end
+
+def my_inject_option2(initial, input, arr)
+  if input == :+
+    memo = initial
+    arr.each do |x|
+      memo += x
+    end
+  elsif input == :-
+    memo = initial
+    arr.each do |x|
+      memo -= x
+    end
+  elsif input == :*
+    memo = initial
+    arr.each do |x|
+      memo *= x
+    end
+  elsif input == :/
+    memo = 0
+    return memo = 1 if arr.my_all?(1)
+  end
+  memo
+end
+
+def my_inject(initial = nil, input = nil)
```

Comment:
- there is a behavior in #my_inject that you need to work on:

` (5..10)).my_inject { |prod, n| prod * n } === (5..10)).inject { |prod, n| prod * n } => false `

Reply:
![proc](https://user-images.githubusercontent.com/65357692/90681939-d26b3800-e229-11ea-90bf-c059c18dbe9c.png)